extensions/sql-migration/src/api/azure.ts

```diff
@@ -58,3 +58,59 @@ export async function getAvailableSqlVMs(account: azdata.Account, subscription:
 	const result = await api.runGraphQuery<SqlVMServer>(account, [subscription], false, `where type == "${azureResource.AzureResourceType.virtualMachines}" and properties.storageProfile.imageReference.publisher == "microsoftsqlserver"`);
 	return result.resources;
 }
+
```

Comment:
Suggest moving the implementation of all these APIs: getAvailableManagedInstanceProducts, getAvailableSqlServers, getAvailableSqlVMs, getAvailableyStorageAccounts, getFileShares, getBlobContainers directly as API methods in azurecore as they expose fetching of very useful artifacts specific to azure, these artifacts can be used by other extension wanting to interact and get these from azure and these are not specific to sql-migration scenario necessarily.

Reply:
The extensions might still have some queries/calls that may be very specific to their workflows. I can probably move these calls to azure core but I would like to expose makeHTTPGetRequest and RunGraphQuery to everyone.